pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/GridDistanceFunction.java

```
@@ -0,0 +1,84 @@
+
+/**
+ * Function that calculates the grid distance between two H3 indexes.
+ * The function takes two arguments:
+ *  - gridDistance(firstH3Index, secondH3Index)
+ */
+public class GridDistanceFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "gridDistance";
+  private TransformFunction _firstArgument;
+  private TransformFunction _secondArgument;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+    super.init(arguments, columnContextMap);
+    Preconditions.checkArgument(arguments.size() == 2,
+        "Transform function %s requires 2 arguments", getName());
+
+    TransformFunction transformFunction = arguments.get(0);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+        "First argument must be single-valued for transform function: %s", getName());
+    _firstArgument = transformFunction;
+
+    transformFunction = arguments.get(1);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+        "Second argument must be single-valued for transform function: %s", getName());
+    _secondArgument = transformFunction;
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return new TransformResultMetadata(FieldSpec.DataType.LONG, true, false);
+  }
+
+  @Override
+  public long[] transformToLongValuesSV(ValueBlock valueBlock) {
+    int numDocs = valueBlock.getNumDocs();
+    initLongValuesSV(numDocs);
+
+    long[] firstH3Indexes = _firstArgument.transformToLongValuesSV(valueBlock);
+    long[] secondH3Indexes = _secondArgument.transformToLongValuesSV(valueBlock);
+
+    for (int i = 0; i < numDocs; i++) {
+      try {
+        _longValuesSV[i] = ScalarFunctions.gridDistance(firstH3Indexes[i], secondH3Indexes[i]);
+      } catch (Exception e) { }
```

Let's not swallow exception. Is there any expected exception?

---

Sounds good. I believe there was in V3, but not any more with `H3_CORE`. Removed the ignore try/catch.